v3/src/components/free-tile-row.tsx

```diff
@@ -14,24 +14,95 @@ interface IFreeTileRowProps {
   getTile: (tileId: string) => ITileModel | undefined
 }
 export const FreeTileRowComponent = observer(({ content, row, getTile }: IFreeTileRowProps) => {
+  const [resizingTileStyle, setResizingTileStyle] =
+    useState<{left: number, top: number, width: number, height: number}>()
+  const [resizingTileId, setResizingTileId] = useState("")
+  const tempWidth = useRef<number>(0)
+  const tempHeight = useRef<number>(0)
+  const tempLeft = useRef<number>(0)
```

Comment:
When I suggested using refs I was thinking of them as an alternative to the state. Since you're primarily using the state values for the same purpose, I think you can probably get rid of the refs and just use the state values throughout.